packages/core/contracts/cross-chain-oracle/ChildMessengerInterface.sol

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+interface ChildMessengerInterface {
+    // Should send cross-chain message to Parent messenger contract or revert.
+    function sendMessageToParent(bytes memory data) external;
+
+    // Should be targeted by ParentMessenger and executed upon receiving a message from root chain.
+    function processMessageFromParent(bytes memory data) external;
```

Comment:
this will happen within the implementer of this interface(I'm working on this PR separately). This will work by using the notion of a "cross domain messenger". The implementation for this will be very similar to how the OVM/AVM bridge deposit boxes only accept calls cross chain from their respective L1 owners. This is restricted using a `onlyFromCrossDomainAccount` type modifier.

See here for the cross domain enabled logic and how you can restrict who the caller is from the other chain (i.e how you limit this to only be called by the _parent_: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/external/ovm/OVM_CrossDomainEnabled.sol 

and here for how this is used within a _messenger_. This is similar to what the cross-chain adapters would look like for this set of contracts: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/insured-bridge/ovm/Optimism_Messenger.sol

Reply:
> this will happen within the implementer of this interface(I'm working on this PR separately). This will work by using the notion of a "cross domain messenger". The implementation for this will be very similar to how the OVM/AVM bridge deposit boxes only accept calls cross chain from their respective L1 owners. This is restricted using a onlyFromCrossDomainAccount type modifier.

Agreed with Chris. In the Hub/Spoke contracts its important to restrict access only to the Messenger contracts. The Messenger contracts then need to make sure they can only communicate with each other. We are assuming that the messenger contracts are trusted, and accordingly only the owner of the Hub/Spoke can set the Messenger contracts

Reply:
So just so I understand, there will be a separate messenger for the governor and for the oracle.

For Optimism for instance, we would have an oracle child messenger, a governor child messenger, an oracle parent messenger, and an oracle child messenger?

I think I had been reading it as a single messenger per chain, but it sounds like we're doing this similar to how to polygon implementation works where each contract has exactly one contract on the other side that it can communicate with.

Reply:
> So just so I understand, there will be a separate messenger for the governor and for the oracle.
I think we'll combine them into one messenger with functions specific to message the Governor/Oracle.

Reply:
The implementation I'm working towards does not have separate methods for the governor and oracle but rather stores the addresses of these contracts to verify the caller has the associated permissions